core/internal/cltest/cltest.go

```diff
@@ -619,6 +619,10 @@ type User struct {
 func (ta *TestApplication) NewHTTPClient(user *User) HTTPClientCleaner {
 	ta.t.Helper()
 
+	if user == nil {
```

Comment:
It looks this API does two different things based on how the consumer uses

1. The consumer passes a `*User` object in
2. The consumer passes `nil` in

Given the updates in the PR, it feels like we should either change the API to accept no user, or create two APIs:
* `NewHTTPClientWithUser`
* `NewHTTPClient`

What do you think?